storage/static.go

```diff
@@ -91,8 +92,9 @@ type staticPasswordsStorage struct {
 func WithStaticPasswords(s Storage, staticPasswords []Password) Storage {
 	passwordsByEmail := make(map[string]Password, len(staticPasswords))
 	for _, p := range staticPasswords {
-		p.Email = strings.ToLower(p.Email)
-		passwordsByEmail[p.Email] = p
+		//Enable case insensitive email comparison.
+		lowerEmail := strings.ToLower(p.Email)
+		passwordsByEmail[lowerEmail] = p
```

Comment:
So if we don't error out we essentially allow the user to overwrite the map. For the following ConfigMap:
```
staticPasswords:
- email: "admin@example.com"
  # bcrypt hash of the string "password"
  hash: "$2a$10$2b2cU8CPhOTaGrs1HRQuAueS7JTT5ZHsHSzYiFPm1leZck7Mc8T4W"
  username: "admin"
  userID: "08a8684b-db88-4b73-90a9-3cd1661f5466"
- email: "Admin@example.com"
  # bcrypt hash of the string "password"
  hash: "$2a$10$2b2cU8CPhOTaGrs1HRQuAueS7JTT5ZHsHSzYiFPm1leZck7Mc8T4W"
  username: "Admin"
  userID: "08a8684b-db88-4b73-90a9-3cd1661f5466"
```

Only the second password gets registered. And then if the user tries to login with the first email it still succeed but returns the wrong email id in the claims:

```
"email": "Admin@example.com",
```

That's not desirable behavior :(